.github/workflows/gh-linux_mac.yml

```diff
@@ -0,0 +1,360 @@
+    timeout-minutes: 90
+    strategy:
+      matrix:
+          os: [ubuntu-16.04, macOS-10.14]
+          config: [S3, GCS, AZURE, HDFS, ASAN, SERIALIZATION]
+          exclude:
+            - {os: macOS-10.14, config: S3}
+            - {os: macOS-10.14, config: AZURE}
+            - {os: macOS-10.14, config: HDFS}
+            - {os: macOS-10.14, config: ASAN}
+            - {os: macOS-10.14, config: SERIALIZATION}
+          include:
+            - config: S3
+              TILEDB_S3: ON
+              TILEDB_STATIC: OFF
+              TILEDB_TOOLS: ON
+              CXX: g++
+            - config: GCS
+              TILEDB_GCS: ON
+              TILEDB_STATIC: OFF
+              CXX: g++
+            - config: HDFS
+              TILEDB_HDFS: ON
+              CXX: g++
+            - config: AZURE
+              TILEDB_AZURE: ON
+              TILEDB_STATIC: OFF
+              TILEDB_ARROW_TESTS: ON
+              CXX: g++
+            - config: ASAN
+              TILEDB_CI_ASAN: ON
+              TILEDB_SERIALIZATION: ON
+              CXX: g++-7
+            - config: SERIALIZATION
+              TILEDB_SERIALIZATION: ON
+              TILEDB_S3: ON
+              CXX: g++
+              BACKWARDS_COMPATIBILITY_ARRAYS: ON
+
+    name: Build - ${{ matrix.os }} - ${{ matrix.config }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: 'Print env'
+        run: |
+          echo "'uname -s' is:"
+          echo "uname: " $(uname)
+          echo "uname -m: " $(uname -m)
+          echo "uname -r:" $(uname -r)
+          echo "uname -s: " $(uname -s)
+          echo "uname -v: " $(uname -v)
+          printenv
+        shell: bash
+
+      # Need this for virtualenv and arrow tests if enabled
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+
+      - run: |
+          set -e pipefail
+          python -m pip install --upgrade pip virtualenv
+          if [[ ${{ matrix.TILEDB_ARROW_TESTS == 'ON' }} ]]; then
+            pip install pyarrow pybind11 numpy
+          fi
+        shell: bash
+
+      - name: 'Install doxygen (linux only)'
+        run: |
+          set -e pipefail
+          # Install doxygen *before* running cmake
+          sudo apt-get install doxygen
+        shell: bash
+        if: ${{ runner.os == 'Linux' }}
+
+      - name: 'Install system headers (OSX 10.14 only)'
+        run: |
+          set -e pipefail
+          open /Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg
+          sudo installer -pkg /Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg -allowUntrusted -target /
+        shell: bash
+        if: ${{ runner.os == 'macOS' && env.imageName == 'macOS-10.14' }}
+
+      - name: Check formatting (linux only)'
+        run: |
+          set -e pipefail
+          # Install clang-format (v5.0)
+          ls -la
+          sudo ./scripts/install-clangformat.sh
```

Comment:
I believe clang-format is available in the images, so we could remove this step and move this up above the other steps, so that it fails as fast as possible. (see https://github.com/actions/virtual-environments/blob/7c6117e6ec7c92e1717864296290fbf4a5c3e8f0/images/linux/scripts/installers/clang.sh)

Reply:
Don't dependent though on what clang-format version the image provides? I get that you can pass it as argument in the script you refer `$1`, but imo wouldn't it better to fully handle it through our script as it is now?

Reply:
We have a format configuration file in `SRC/.clang-format`, so I think the output should be pretty stable across versions. I run `clang-format version 11.0.0` locally and have not had any conflicts. The reason I suggest to use the version in the image is to avoid the install step, which looks like it takes ~20 out of 22s for this step. 

That said, marking as resolved for now so it is 1:1 comparable with the AZP pipeline. After merging we should probably move this out to a separate standalone, prerequisite step because it really only needs to run once (can avoid firing off all the jobs just to have GHA kill them when one check-format step completes with error).